packages/adapters/src/kubernetes-instance-adapter.ts

```diff
@@ -0,0 +1,171 @@
+import { adapterConfigDecoder } from "./kubernetes-config-decoder";
+
+/**
+ * Adapter for running Instance by Runner executed in separate process.
+ */
+class KubernetesInstanceAdapter implements
+ILifeCycleAdapterMain,
+ILifeCycleAdapterRun,
+IComponent {
+    logger: IObjectLogger;
+    name = "KubernetesInstanceAdapter"
+
+    private _runnerName?: string
+    private _kubeClient?: KubernetesClientAdapter;
+
+    private adapterConfig: K8SAdapterConfiguration
+
+    constructor(sthConfig: STHConfiguration) {
+        // @TODO this is a redundant check (it was already checked in sequence adapter)
+        // We should move this to config service decoding: https://github.com/scramjetorg/transform-hub/issues/279
+        const decodedAdapterConfig = adapterConfigDecoder.decode(sthConfig.kubernetes);
+
+        if (!decodedAdapterConfig.isOk()) {
+            throw new Error("Invalid Kubernetes Adapter configuration");
+        }
+
+        this.adapterConfig = decodedAdapterConfig.value;
+        this.logger = new ObjLogger(this.name);
+    }
+
+    private get kubeClient() {
+        if (!this._kubeClient) {
+            throw new Error("Kubernetes client not initialized");
+        }
+
+        return this._kubeClient;
+    }
+
+    async init(): Promise<void> {
+        this._kubeClient = new KubernetesClientAdapter(this.adapterConfig.authConfigPath, this.adapterConfig.namespace);
+        this.kubeClient.init();
+
+        this._kubeClient.logger.pipe(this.logger);
+    }
+
+    async stats(msg: MonitoringMessageData): Promise<MonitoringMessageData> {
+        return {
+            ...msg,
+        };
+    }
+
+    async run(config: SequenceConfig, instancesServerPort: number, instanceId: string): Promise<ExitCode> {
+        if (config.type !== "kubernetes") {
+            throw new Error(`Invalid config type for kubernetes adapter: ${config.type}`);
+        }
+
+        const runnerName = this._runnerName = `runner-${ instanceId }`;
+
+        this.logger.debug("Creating Runner Pod");
+
+        const env = Object.entries({
+            SEQUENCE_PATH: path.join("/package", config.entrypointPath),
+            DEVELOPMENT: process.env.DEVELOPMENT ?? "",
+            PRODUCTION: process.env.PRODUCTION ?? "",
+            INSTANCES_SERVER_PORT: instancesServerPort.toString(),
+            INSTANCES_SERVER_HOST: this.adapterConfig.sthPodHost,
+            INSTANCE_ID: instanceId,
+        }).map(([name, value]) => ({ name, value }));
+
+        await this.kubeClient.createPod(
+            {
+                name: runnerName,
+                labels: {
+                    app: "runner"
+                }
+            },
+            {
+                containers: [{
+                    env,
+                    name: runnerName,
+                    image: this.adapterConfig.runnerImage,
+                    stdin: true,
+                    command: ["wait-for-sequence-and-start.sh"],
+                    imagePullPolicy: "Always"
```

Comment:
We should remove that or hide it under DEVELOPMENT env var

Reply:
Added to https://github.com/scramjetorg/transform-hub/issues/268.